src/core/Akka.Remote/Endpoint.cs

```diff
@@ -256,59 +263,71 @@ public OversizedPayloadException(string msg)
 
     /// <summary>
     /// INTERNAL API
-    /// 
-    /// <remarks>
-    /// [Aaronontheweb] so this class is responsible for maintaining a buffer of retriable messages in
-    /// Akka and it expects an ACK / NACK response pattern before it considers a message to be sent or received.
-    /// 
-    /// Currently AkkaDotNet does not have any form of guaranteed message delivery in the stack, since that was
-    /// considered outside the scope of V1. However, this class needs to be revisited and updated to support it,
-    /// along with others.
-    /// 
-    /// For the time being, the class remains just a proxy for spawning <see cref="EndpointWriter"/> actors and
-    /// forming any outbound associations.
-    /// </remarks>
     /// </summary>
-    internal class ReliableDeliverySupervisor : UntypedActor
+    internal class ReliableDeliverySupervisor : ReceiveActor
     {
+        #region Internal message classes
```

Comment:
These are the new messages to support the `Idle` state that the `ReliableDeliverySupervisor` can be in.